import "type.proto";
package pbnavitia.stat;
	
message StatParameter {
    optional string key = 1;
    optional string value = 2;    
}

message Coord{
    optional double lat = 1;
    optional double lon = 2;
}

message StatCoverage{
    optional string region_id = 1;
}

message StatRequest {
    //POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC)    
    optional uint64	request_date = 1;
    optional int32	user_id = 2;
    optional string	user_name = 3;
    optional int32	application_id = 4;
    optional string	application_name = 5;
    optional int32	request_duration = 6;
    optional string	api = 7;
    optional string   path = 8;
    optional string	query_string = 9;
    optional string	host = 10;
    optional int32	response_size = 11;
    optional string	client = 12;
    repeated StatCoverage coverages = 13;    
    repeated StatParameter parameters = 14;
    repeated StatJourneyResponse journeys = 15;
}

message StatJourneyResponse {
    optional uint64 	requested_date_time = 1;
    optional uint64 	departure_date_time = 2;
    optional uint64 	arrival_date_time = 3;
    optional int32 	duration = 4;
    optional int32 	nb_transfers = 5;
    optional string 	type = 6;
    repeated StatJourneySection sections = 7;
}

message StatJourneySection {
    optional uint64 	departure_date_time = 1;
    optional uint64 	arrival_date_time = 2;
    optional int32 	duration = 3;
    optional string 	type = 4;
    optional string   mode = 5;
    optional string 	from_embedded_type = 6;
    optional string 	from_id = 7;
    optional Coord 	from_coord = 8;
    optional string 	to_embedded_type = 9;
    optional string 	to_id = 10;
    optional Coord 	to_coord = 11;
    optional string 	vehicle_journey_id = 12;
    optional string 	line_id = 13;
    optional string 	route_id = 14;
    optional string 	network_id = 15;
    optional string 	physical_mode_id = 16;
    optional string 	commercial_mode_id = 17;
}


